packages/web3-eth/types/index.d.ts

```diff
@@ -220,3 +222,17 @@ export interface Subscription<T> {
 
     on(type: 'error', handler: (data: Error) => void): void
 }
+
+export interface GetProof {
+    jsonrpc: string;
+    id: number;
+    result: {
+      address: string;
+      accountProof: string[];
+      balance: number | string;
```

Comment:
i think this will always come back as a hex string `0x23` so i think string is fine.